### PR TITLE
#1338 - Added refactored process definition factory for auth checks, …

### PR DESCRIPTION
**Pull request by badvision**
…broadened allowed groups for asset ingestion

This is a small refactoring, but wanted to run the idea by you since it supposes group names which might not exist (but it's not going to fail or throw errors if those groups do not exist)

### CODE REVIEW

File: bundle/src/main/java/com/adobe/acs/commons/mcp/AdministratorsOnlyProcessDefinitionFactory.java [hunk 1]
```diff
@@ -20,36 +20,17 @@
...
-                        try {
-                            return g.getID().equals("administrators");
-                        } catch (RepositoryException e) {
-                            return false;
-                        }
-                });
-        } catch (RepositoryException e) {
-            return false;
-        }
+    public String[] getAuthorizedGroups() {
```
Comment (justinedelson) [inline bundle/src/main/java/com/adobe/acs/commons/mcp/AdministratorsOnlyProcessDefinitionFactory.java]: this method should probably be final. Also it is missing the @Override annotation AFAICT

File: bundle/src/main/java/com/adobe/acs/commons/mcp/AuthorizedGroupProcessDefinitionFactory.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                                } catch (RepositoryException ex) {
+                                    return false;
+                                }
+                            });
+        } catch (RepositoryException e) {
+            return false;
+        }
+    }
+
+    public abstract String[] getAuthorizedGroups();
```
Comment (justinedelson) [inline bundle/src/main/java/com/adobe/acs/commons/mcp/AuthorizedGroupProcessDefinitionFactory.java]: can be protected scope

File: bundle/src/main/java/com/adobe/acs/commons/mcp/AdministratorsOnlyProcessDefinitionFactory.java (hunk 1 above)
Comment (badvision) [reply bundle/src/main/java/com/adobe/acs/commons/mcp/AdministratorsOnlyProcessDefinitionFactory.java]: It has the annotation but making it final makes sense.

File: bundle/src/main/java/com/adobe/acs/commons/mcp/AuthorizedGroupProcessDefinitionFactory.java (hunk 2 above)
Comment (badvision) [reply bundle/src/main/java/com/adobe/acs/commons/mcp/AuthorizedGroupProcessDefinitionFactory.java]: That is true in a few other places too.  Pushing an update for access modifiers.

File: bundle/src/main/java/com/adobe/acs/commons/mcp/AuthorizedGroupProcessDefinitionFactory.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ * groups.
+ *
+ * @param <P> Process definition generated by this factory
+ */
+@ConsumerType
+public abstract class AuthorizedGroupProcessDefinitionFactory<P extends ProcessDefinition> extends ProcessDefinitionFactory<P> {
+
+    @Override
+    @SuppressWarnings("squid:S1141")
+    public boolean isAllowed(User user) {
```
[the remark this answers is not in the document]
Comment (badvision) [reply bundle/src/main/java/com/adobe/acs/commons/mcp/AuthorizedGroupProcessDefinitionFactory.java]: I refactored the existing administrator implementation to use this and also added additional groups allowed to use the asset ingest tools.

**Comment by davidjgonzalez** [comment]
@badvision you planning on fixing JE's 2 comments?

**Comment by badvision** [comment]
Yes sir, already done.

**Pull request merged**